GliaWidgets/Window/GliaViewController.swift

```diff
@@ -0,0 +1,107 @@
+import UIKit
+
+public enum GliaViewControllerEvent {
+    case minimized
+    case maximized
+}
+
+public protocol GliaViewControllerDelegate: AnyObject {
+    func event(_ event: GliaViewControllerEvent)
+}
+
+class GliaViewController: UIViewController {
+    var bubbleKind: BubbleKind = .userImage(url: nil) {
+        didSet { bubbleWindow?.bubbleKind = bubbleKind }
+    }
+
+    private enum State {
+        case maximized
+        case minimized
+    }
+
+    private var state: State = .maximized
+    private weak var delegate: GliaViewControllerDelegate?
+    private var gliaNavigationController: UINavigationController!
+    private let bubbleView: BubbleView
+    private var bubbleWindow: BubbleWindow?
+    private var animationImageView: UIImageView?
+
+    private var maximizeScreenshot: UIImage? {
+        UIApplication.shared.windows.first?.screenshot
+    }
+
+    init(bubbleView: BubbleView, delegate: GliaViewControllerDelegate?) {
+        self.bubbleView = bubbleView
+        self.delegate = delegate
+        super.init(nibName: nil, bundle: nil)
+        setup()
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func insertNavigationController(_ navigationController: UINavigationController) {
+        self.gliaNavigationController = navigationController
+        addChild(navigationController)
+        view.addSubview(navigationController.view)
+        navigationController.didMove(toParent: self)
+    }
```

Comment:
Probably we could extend `UIViewController` class and move this method to separated extension file like this:

```suggestion
    func add(_ viewController: UIViewController) {
        viewController.willMove(toParent: self)
        addChild(viewController)
        view.addSubview(viewController.view)
        viewController.didMove(toParent: self)
    }
```

One more question about layout, does it use resizing mask or layout constraints?

Reply:
The thought of moving it into an extension hasn't crossed my mind, to be honest, but why not.
As for the layout, it's using constraints.

Reply:
Named the extension function `insertChild`, because just `add` seemed not verbose enough to me.